features/poll/impl/src/main/kotlin/io/element/android/features/poll/impl/create/CreatePollPresenter.kt

```diff
@@ -0,0 +1,168 @@
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.saveable.Saver
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import io.element.android.libraries.architecture.Presenter
+import io.element.android.libraries.matrix.api.poll.PollKind
+import io.element.android.libraries.matrix.api.room.MatrixRoom
+import io.element.android.services.analytics.api.AnalyticsService
+import kotlinx.collections.immutable.ImmutableList
+import kotlinx.collections.immutable.toImmutableList
+import kotlinx.coroutines.launch
+import timber.log.Timber
+
+private const val MIN_ANSWERS = 2;
+private const val MAX_ANSWERS = 20;
+private const val MAX_ANSWER_LENGTH = 240;
+private const val MAX_SELECTIONS = 1;
+
+class CreatePollPresenter @AssistedInject constructor(
+    private val room: MatrixRoom,
+    private val analyticsService: AnalyticsService,
+    @Assisted private val navigateUp: () -> Unit,
+    // private val messageComposerContext: MessageComposerContext, // TODO
+) : Presenter<CreatePollState> {
+
+    @AssistedFactory
+    interface Factory {
+        fun create(backNavigator: () -> Unit): CreatePollPresenter
+    }
+
+    @Composable
+    override fun present(): CreatePollState {
+
+        var question: String by rememberSaveable { mutableStateOf("") }
+        var answers: List<String> by rememberSaveable() { mutableStateOf(listOf("", "")) }
+        var pollKind: PollKind by rememberSaveable(saver = pollKindSaver) { mutableStateOf(PollKind.Disclosed) }
+        var showConfirmation: Boolean by rememberSaveable { mutableStateOf(false) }
+
+        val canCreate: Boolean by remember { derivedStateOf { canCreate(question, answers) } }
+        val canAddAnswer: Boolean by remember { derivedStateOf { canAddAnswer(answers) } }
+        val immutableAnswers: ImmutableList<Answer> by remember { derivedStateOf { answers.toAnswers() } }
+
+        val scope = rememberCoroutineScope()
+
+        fun handleEvents(event: CreatePollEvents) {
+            when (event) {
+                is CreatePollEvents.Create -> scope.launch {
+                    if (canCreate) {
+                        room.createPoll(
+                            question = question,
+                            answers = answers,
+                            maxSelections = MAX_SELECTIONS,
+                            pollKind = pollKind,
+                        )
+                        // analyticsService.capture(PollCreate()) // TODO: Send PollCreate analytics.
+                        navigateUp()
+                    } else {
+                        Timber.e("Cannot create poll")
```

Comment:
I guess this else block can be removed, to log is not adding lots of value. Or just change it to a debug log.